### PR TITLE
Added support for ALTER TABLE t REORGANIZE PARTITION p INTO (..)

### LINKED ISSUE #6444: Support ALTER TABLE t REORGANIZE PARTITION <list of partitions> INTO (<list of partition definitions>)
## Enhancement

See https://github.com/pingcap/tidb/issues/38535 for more info about the feature.

TiFlash needs support for intermediate partitions, during the partition data reorg, where new the data from the reorganized partitions are copied to intermediate partitions, that will finally replace the original reorganized partitions.

This means that there will be multiple partitions that are active in TiFlash during the operation, since there will be possible to read from both the new and the old depending on schema version of the client.

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Reference: https://github.com/pingcap/tidb/issues/38535
Issue Number: close #6444 

Problem Summary:
Support ALTER TABLE t REORGANIZE PARTITION <list of partitions> INTO (<list of partition definitions>)

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hongyunyan** [comment]
It's better to add some Integration tests under tests/fullstack-test2/ddl to make verify.

**Comment by JaySon-Huang** [comment]
@mjonss report that he see this error `ERROR 1105 (HY000): [FLASH:Coprocessor:RegionError] Region 99 is unavailable at 16: (while doing learner read for table, logical table_id: 80)` when testing `REORGANIZE PARTITION` with this tiflash.
@hongyunyan Can you help @mjonss to verify the `REORGANIZE PARTITION` works well before getting this PR merged?

**Comment by hongyunyan** [comment]
> @mjonss report that he see this error `ERROR 1105 (HY000): [FLASH:Coprocessor:RegionError] Region 99 is unavailable at 16: (while doing learner read for table, logical table_id: 80)` when testing `REORGANIZE PARTITION` with this tiflash. @hongyunyan Can you help @mjonss to verify the `REORGANIZE PARTITION` works well before getting this PR merged?

No problem！While for this error @mjonss said he got that error even before running the new ALTER TABLE t REORGANIZE PARTITION.

File: dbms/src/TiDB/Schema/SchemaGetter.h [hunk 1]
```diff
@@ -95,11 +95,12 @@ enum class SchemaActionType : Int8
     AlterNoCacheTable = 59,
     CreateTables = 60,
     ActionMultiSchemaChange = 61,
+    AlterTableReorganizePartition = 64,
```
[the remark this answers is not in the document]
Comment (mjonss) [reply dbms/src/TiDB/Schema/SchemaGetter.h:100]: Updated

**Comment by mjonss** [comment]
> > @mjonss report that he see this error `ERROR 1105 (HY000): [FLASH:Coprocessor:RegionError] Region 99 is unavailable at 16: (while doing learner read for table, logical table_id: 80)` when testing `REORGANIZE PARTITION` with this tiflash. @hongyunyan Can you help @mjonss to verify the `REORGANIZE PARTITION` works well before getting this PR merged?
> 
> No problem！While for this error @mjonss said he got that error even before running the new ALTER TABLE t REORGANIZE PARTITION.

Seems related to arm64?
I created https://github.com/pingcap/tiflash/issues/6659.

**Comment by mjonss** [comment]
@hongyunyan Hi, the [Reorganize partition](https://github.com/pingcap/tidb/pull/41096) feature in TiDB repo is in the merge queue, is it possible to review and merge this as well now?

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by hongyunyan — APPROVED** [review]
LGTM

**Pull request merged**